### ISSUE TITLE
requires_grad be False after encoding

### ISSUE THREAD

**Issue by Kyrie-Irving07** [NVlabs/tiny-cuda-nn#77]
Thanks for the great project! 
I encountered a problem when applying backward function. Specifically, the requires_grad of the outputs from the encoder would be False. I just want to calculate the gradients of the sigma value to the coordinates to estimate the normals. 

```
import torch
import torch.nn.functional as F
import tinycudann as tcnn

sh_enc_degree = 4

sh_encoder = tcnn.Encoding(
    n_input_dims=3,
    encoding_config={"otype": "HashGrid", "degree": sh_enc_degree},
    dtype=torch.float32,
)

inputs = F.normalize(torch.randn(100, 3), dim=-1).cuda()
inputs.requires_grad_()
outputs = sh_encoder(inputs)
outputs.requires_grad
```

**Comment by Kyrie-Irving07** [comment]
Solved by commenting 
'with torch.no_grad():'

**Comment by Tom94** [comment]
I'm afraid I don't understand how the problem happened in the first place. May I ask in which context you had to comment `with torch.no_grad():`? Was it in __tcnn__'s `modules.py`?

**Comment by Kyrie-Irving07** [comment]
`with torch.no_grad():` is in my codes. **tcnn** is fine. Sorry for bothering.
Thanks again for the great project!
